src/components/TextArea.tsx

```diff
@@ -0,0 +1,100 @@
+import React, {
+  FunctionComponent,
+  InputHTMLAttributes,
+  ReactNode,
+} from "react";
+
+interface TextAreaProps {
+  name: string;
+  id?: string;
+  htmlAttributes?: InputHTMLAttributes<Element>;
+  rows?: number;
+}
+
+interface TextAreaCharacterCountProps {
+  name: string;
+  id: string;
+  maxCharacters: number;
+  rows?: number;
+  htmlAttributes?: InputHTMLAttributes<Element>;
+  children: ReactNode;
+}
+
+export const TextArea: FunctionComponent<TextAreaProps> = ({
+  id = null,
+  name,
+  htmlAttributes = {},
+  rows = 3,
+}: TextAreaProps): JSX.Element => {
+  return (
+    <textarea
+      className="govuk-textarea"
+      id={id}
+      name={name}
+      rows={rows}
+      aria-describedby={`${id}-hint`}
+      {...htmlAttributes}
+    />
+  );
+};
+
+export const TextAreaCharacterCount: FunctionComponent<TextAreaCharacterCountProps> = ({
+  name,
+  id,
+  maxCharacters,
+  htmlAttributes,
+  rows,
+  children,
+}: TextAreaCharacterCountProps): JSX.Element => {
+  const [count] = React.useState(0);
+
+  const withinCharacterLimit = count <= maxCharacters;
+
+  const textAreaClass = () => {
+    return withinCharacterLimit
+      ? "govuk-textarea govuk-js-character-count"
+      : "govuk-textarea govuk-js-character-count govuk-textarea--error";
+  };
+
+  const textAreaErrorAria = !withinCharacterLimit && `${id}-error`;
+
+  const characterCountClass = () => {
+    return withinCharacterLimit
+      ? "govuk-hint govuk-character-count__message"
+      : "govuk-error-message govuk-character-count__message";
+  };
+  const characterCountMessage = () => {
+    if (count === 0) {
+      return `You can enter up to ${maxCharacters} characters`;
+    } else {
+      return withinCharacterLimit
+        ? `You have ${maxCharacters - count} characters remaining`
+        : `You have ${count - maxCharacters} characters too many`;
+    }
+  };
```

Comment:
Nice work 👍 

Looking into the [govuk JS](https://github.com/alphagov/govuk-frontend/blob/master/src/govuk/components/character-count/character-count.js) I don't think we actually have to do any of the JS ourselves, it should just be a case of adding the correct `data-*` attributes and class names.

Have you able to just use the [HTML](https://design-system.service.gov.uk/components/character-count/) from the character component from the GOV UK and get the character count working?

Reply:
Yup!

Initially, I was trying to Make TextAreaCharacterCount be more of a wrapper around TextArea, so that TextArea wasn't being duplicated. And needed to add some JS to that for the character count to work.

But then that got messy, so when I just made TextAreaCharacterCount its own thing (how it is now) and I think I applied the correct classes, since I copied over the exact thing in the Gov UK website, and then the JS just worked on its own!

Reply:
So I didn't need any of this JS, because GovUk frontend is magical and does it all 😭